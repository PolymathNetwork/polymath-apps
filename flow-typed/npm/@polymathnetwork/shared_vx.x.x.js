// flow-typed signature: 4bd024aac82bd9d25d925fc8faf72b97
// flow-typed version: <<STUB>>/@polymathnetwork/shared_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/shared'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/shared' {
  declare module.exports: any;
}
