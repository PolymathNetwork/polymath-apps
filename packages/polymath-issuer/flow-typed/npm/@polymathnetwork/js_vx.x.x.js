// flow-typed signature: f74591e4e08edd90ce9c87e65e769c1d
// flow-typed version: <<STUB>>/@polymathnetwork/js_v0.0.0/flow_v0.81.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@polymathnetwork/js'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@polymathnetwork/js' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@polymathnetwork/js/babel.config' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CappedSTOFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/Contract' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CountTransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/CountTransferManagerFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/IModuleFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PercentageTransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PermissionManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/PolyToken' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/SecurityToken' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/SecurityTokenRegistry' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/STO' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/TickerRegistry' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/contracts/TransferManager' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/index' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/IPFS' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/src/types' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/webpack.dev' {
  declare module.exports: any;
}

declare module '@polymathnetwork/js/webpack.prod' {
  declare module.exports: any;
}

// Filename aliases
declare module '@polymathnetwork/js/babel.config.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/babel.config'>;
}
declare module '@polymathnetwork/js/src/contracts/CappedSTOFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CappedSTOFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/Contract.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/Contract'>;
}
declare module '@polymathnetwork/js/src/contracts/CountTransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CountTransferManager'>;
}
declare module '@polymathnetwork/js/src/contracts/CountTransferManagerFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/CountTransferManagerFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/IModuleFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/IModuleFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/PercentageTransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PercentageTransferManager'>;
}
declare module '@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PercentageTransferManagerFactory'>;
}
declare module '@polymathnetwork/js/src/contracts/PermissionManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PermissionManager'>;
}
declare module '@polymathnetwork/js/src/contracts/PolyToken.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/PolyToken'>;
}
declare module '@polymathnetwork/js/src/contracts/SecurityToken.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/SecurityToken'>;
}
declare module '@polymathnetwork/js/src/contracts/SecurityTokenRegistry.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/SecurityTokenRegistry'>;
}
declare module '@polymathnetwork/js/src/contracts/STO.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/STO'>;
}
declare module '@polymathnetwork/js/src/contracts/TickerRegistry.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/TickerRegistry'>;
}
declare module '@polymathnetwork/js/src/contracts/TransferManager.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/contracts/TransferManager'>;
}
declare module '@polymathnetwork/js/src/index.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/index'>;
}
declare module '@polymathnetwork/js/src/IPFS.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/IPFS'>;
}
declare module '@polymathnetwork/js/src/types.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/src/types'>;
}
declare module '@polymathnetwork/js/webpack.dev.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/webpack.dev'>;
}
declare module '@polymathnetwork/js/webpack.prod.js' {
  declare module.exports: $Exports<'@polymathnetwork/js/webpack.prod'>;
}
