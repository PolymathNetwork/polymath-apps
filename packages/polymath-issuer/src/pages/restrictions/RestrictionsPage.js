import React, { Component } from 'react';
import {
  addressShortifier,
  confirm,
  ico24H,
  etherscanAddress,
  NotFoundPage,
  Page,
  Grid,
  Remark,
  RadioInput,
  FormItem,
} from '@polymathnetwork/ui';
import { connect } from 'react-redux';
import {
  Heading,
  CardFeatureState,
  icons,
  ThemeProvider,
  ButtonLarge,
} from '@polymathnetwork/new-ui';
import { Toggle, Tabs, Tab, Form } from 'carbon-components-react';
import { toggleRestrictions } from '../../actions/restrictions';
import { withFormik } from 'formik';
import GlobalRestrictionsForm from './components/GlobalRestrictionsForm';
import IndividualRestrictions from './components/IndividualRestrictions';
import FormModal from './components/FormModal';
import RestrictionDetails from './components/RestrictionDetails';
import './style.scss';
import {
  addVolumeRestrictionModule,
  getVolumeRestrictionModule,
  archiveVolumeRestrictionModule,
} from '../../actions/restrictions';

class RestrictionsPage extends Component {
  state = {
    isFormModalOpen: false,
    restrictionType: null,
  };

  componentWillMount() {
    this.props.getVolumeRestrictionModule();
  }

  handleOpen = restrictionType => {
    this.setState({ isFormModalOpen: true, restrictionType: restrictionType });
  };

  handleClose = () => {
    this.setState({ isFormModalOpen: false });
  };

  handleToggleRestrictions = async (isToggled: boolean) => {
    const {
      archiveVolumeRestrictionModule,
      addVolumeRestrictionModule,
    } = this.props;
    if (isToggled) {
      addVolumeRestrictionModule();
    } else {
      archiveVolumeRestrictionModule();
    }
  };

  render() {
    const {
      token,
      isRestrictionsToggled,
      dailyRestrictionModified,
      defaultRestrictionModified,
      dailyRestriction,
      defaultRestriction,
    } = this.props;
    const { isFormModalOpen, restrictionType } = this.state;
    if (!token || !token.address) {
      return <NotFoundPage />;
    }
    return (
      <ThemeProvider>
        <Page title="Compliance â€“ Polymath">
          <h1 className="pui-h1">Trading Volume Restrictions</h1>
          <Grid>
            <Grid.Row>
              <Grid.Col gridSpan={6}>
                <h3 className="pui-h3">
                  Allows you to define the maximum number of tokens your
                  investors can sell during a customized rolling period. Limits
                  can be set globally for all investors or individually by
                  investor. For example, you may limit all your investors to
                  selling 100 tokens in a rolling 24-hour period and 1,000
                  tokens in a rolling 90-day period.
                </h3>
                <Remark title="Note">
                  Investors with individual investor limits are exempt from the
                  global limits.
                </Remark>
              </Grid.Col>
              <Grid.Col gridSpan={6}>
                <CardFeatureState
                  status={isRestrictionsToggled ? 'idle' : 'inactive'}
                  IconAsset={icons.SvgRestriction}
                >
                  <Heading color="primary" mt={2}>
                    Enable Enforcement of Trade Volume Restrictions
                  </Heading>
                  <div className="center">
                    <Toggle
                      onToggle={this.handleToggleRestrictions}
                      toggled={isRestrictionsToggled}
                      id="volumeRestrictionToggle"
                    />
                  </div>
                </CardFeatureState>
              </Grid.Col>
            </Grid.Row>
            <Grid.Row>
              <Grid.Col gridSpan={12}>
                {isRestrictionsToggled && (
                  <Tabs>
                    <Tab label="Global Restrictions">
                      <h1 className="pui-h1">Global Restrictions</h1>
                      <h3 className="pui-h3">
                        Specify a single rolling period restriction for all
                        investors. The Global Restriction will enforce a maximum
                        number of tokens each investor can sell during the
                        rolling period you define. Easily set to 24 hours or
                        customize your rolling period. Your trade volume cap is
                        specified as the number or tokens or the percentage of
                        total token supply.
                      </h3>
                      <FormModal
                        isOpen={isFormModalOpen}
                        handleClose={this.handleClose}
                        restrictionType={restrictionType}
                      />
                      <Grid>
                        <Grid.Row>
                          <Grid.Col gridSpan={4}>
                            <CardFeatureState
                              style={{ height: '400px' }}
                              maxWidth="none"
                              status={
                                dailyRestrictionModified ? 'idle' : 'inactive'
                              }
                              IconAsset={icons.Svg24H}
                            >
                              <Heading color="primary" mt={2}>
                                24h Rolling Period Restriction
                              </Heading>
                              <p className="card-text">
                                Configure a maximum number of tokens any
                                investor may be able to sell within a rolling
                                24h period.
                              </p>
                              {dailyRestrictionModified && (
                                <RestrictionDetails
                                  restriction={dailyRestriction}
                                />
                              )}

                              {/* Empty div with height instead of CSS */}
                              <ButtonLarge
                                className="card-button"
                                onClick={() => this.handleOpen('24h')}
                              >
                                Configure 24h Restriction
                              </ButtonLarge>
                            </CardFeatureState>
                          </Grid.Col>
                          <Grid.Col gridSpan={4}>
                            <CardFeatureState
                              style={{ height: '400px' }}
                              maxWidth="none"
                              status={
                                defaultRestrictionModified ? 'idle' : 'inactive'
                              }
                              IconAsset={icons.SvgCalendarOutline}
                            >
                              <Heading color="primary" mt={2}>
                                Custom Rolling Period Restriction
                              </Heading>
                              <p className="card-text">
                                Configure a maximum number of tokens any
                                Investor may be able to sell within a custom
                                rolling period.
                              </p>
                              {defaultRestrictionModified && (
                                <RestrictionDetails
                                  restriction={defaultRestriction}
                                />
                              )}
                              <ButtonLarge
                                className="card-button"
                                onClick={() => this.handleOpen('custom')}
                              >
                                Configure Custom Restriction
                              </ButtonLarge>
                            </CardFeatureState>
                          </Grid.Col>
                        </Grid.Row>
                      </Grid>
                    </Tab>
                    <Tab label="Individual Restrictions">
                      <IndividualRestrictions />
                    </Tab>
                  </Tabs>
                )}
              </Grid.Col>
            </Grid.Row>
          </Grid>
        </Page>
      </ThemeProvider>
    );
  }
}

const mapStateToProps = state => ({
  token: state.token.token,
  isRestrictionsToggled: state.restrictions.isToggled,
  dailyRestrictionModified: state.restrictions.dailyRestrictionModified,
  defaultRestrictionModified: state.restrictions.defaultRestrictionModified,
  dailyRestriction: state.restrictions.dailyRestriction,
  defaultRestriction: state.restrictions.defaultRestriction,
});

const mapDispatchToProps = {
  toggleRestrictions,
  addVolumeRestrictionModule,
  getVolumeRestrictionModule,
  archiveVolumeRestrictionModule,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RestrictionsPage);
