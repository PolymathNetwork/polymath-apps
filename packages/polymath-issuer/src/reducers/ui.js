// @flow
import { LOCATION_CHANGE } from 'connected-react-router';

export type UIState = {
  header: {
    variant: 'default' | 'transparent',
  },
  footer: {
    variant: 'default' | 'transparent',
  },
};

const defaultState: UIState = {
  header: {
    variant: 'default',
  },
  footer: {
    variant: 'default',
  },
};

export default (state: UIState = defaultState, action) => {
  if (action.type === LOCATION_CHANGE) {
    const pathname = action.payload.location.pathname;

    if (pathname === '/') {
      return {
        ...state,
        header: {
          variant: 'transparent',
        },
        footer: {
          variant: 'transparent',
        },
      };
    }

    return defaultState;
  } else {
    return state;
  }
};
