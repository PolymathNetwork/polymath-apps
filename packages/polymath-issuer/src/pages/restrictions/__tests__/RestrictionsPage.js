import { RestrictionsPage } from '../RestrictionsPage';
import DefaultRestrictionsPage from '../RestrictionsPage';
import renderer from 'react-test-renderer';
import React from 'react';
import configureStore from 'redux-mock-store';
import { theme, GlobalStyles } from '@polymathnetwork/ui';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import moment from 'moment';
import { Provider } from 'react-redux';
const mockStore = configureStore([]);

describe('Restrictions Page', () => {
  it('should render not found when there is no token', () => {
    const props = {
      getVolumeRestrictionModule: jest.fn(),
      theme,
    };
    const tree = shallow(<RestrictionsPage {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render the page with VRTM disabled', () => {
    const props = {
      getVolumeRestrictionModule: jest.fn(),
      token: {
        address: '0x121212',
      },
      theme,
    };
    const tree = shallow(<RestrictionsPage {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render global restrictions tab when VRTM is enabled', () => {
    const props = {
      getVolumeRestrictionModule: jest.fn(),
      isRestrictionsToggled: true,
      token: {
        address: '0x121212',
      },
      theme,
    };
    const tree = shallow(<RestrictionsPage {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render global default restrictions when VRTM is enabled', () => {
    const props = {
      getVolumeRestrictionModule: jest.fn(),
      dailyRestrictionModified: true,
      dailyRestriction: {
        startTime: moment(1),
        endTime: moment(1),
        allowedTokens: 500,
        restrictionType: 0,
      },
      isRestrictionsToggled: true,
      token: {
        address: '0x121212',
      },
      theme,
    };
    const tree = shallow(<RestrictionsPage {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render global daily restrictions when VRTM is enabled', () => {
    const props = {
      getVolumeRestrictionModule: jest.fn(),
      defaultRestrictionModified: true,
      defaultRestriction: {
        startTime: moment.unix(1),
        endTime: moment(1),
        allowedTokens: 500,
        restrictionType: 0,
      },
      isRestrictionsToggled: true,
      token: {
        address: '0x121212',
      },
      theme,
    };
    const store = mockStore({
      token: {
        address: '0x121212',
      },
    });
    const tree = shallow(<RestrictionsPage {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
