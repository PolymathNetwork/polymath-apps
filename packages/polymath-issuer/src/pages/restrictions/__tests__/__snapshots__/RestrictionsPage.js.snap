// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restrictions Page should render global daily restrictions when VRTM is enabled 1`] = `
<ThemeProvider>
  <Page
    py="xl"
    title="Compliance – Polymath"
  >
    <h1
      className="pui-h1"
    >
      Trade Volume Restrictions
    </h1>
    <Grid
      gridGap={4}
    >
      <GridRow
        gridGap={4}
      >
        <GridCol
          gridSpan={6}
        >
          <h3
            className="pui-h3"
          >
            Allows you to define the maximum number of tokens your investors can sell during a customized rolling period. Limits can be set globally for all investors or individually by investor. For example, you may limit all your investors to selling 100 tokens in a rolling 24-hour period and 1,000 tokens in a rolling 90-day period.
          </h3>
          <Remark
            title="Note"
          >
            Investors to whom individual limits are applied are only subject to their individual limits and are exempt from the global limits.
          </Remark>
        </GridCol>
        <GridCol
          gridSpan={6}
        >
          <CardFeatureState
            IconAsset={[Function]}
            maxWidth={355}
            status="idle"
            style={Object {}}
          >
            <Heading__StyledHeading
              as="h2"
              color="primary"
              lineHeight="normal"
              mb="m"
              mt={2}
              variant="h3"
            >
              Enable Enforcement of Trade Volume Restrictions
            </Heading__StyledHeading>
            <div
              className="center"
            >
              <Toggle
                defaultToggled={false}
                id="volumeRestrictionToggle"
                labelA="Off"
                labelB="On"
                onToggle={[Function]}
                toggled={true}
              />
            </div>
          </CardFeatureState>
        </GridCol>
      </GridRow>
      <GridRow
        gridGap={4}
      >
        <GridCol
          gridSpan={12}
        >
          <Tabs
            ariaLabel="listbox"
            iconDescription="show menu options"
            role="navigation"
            selected={0}
            triggerHref="#"
          >
            <Tab
              href="#"
              label="Global Restrictions"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={false}
              tabIndex={0}
            >
              <h1
                className="pui-h1"
              >
                Global Restrictions
              </h1>
              <h3
                className="pui-h3"
              >
                Specify a single rolling period restriction for all investors. The Global Restriction will enforce a maximum number of tokens each investor can sell during the rolling period you define. Easily set to 24 hours or customize your rolling period. Your trade volume cap is specified as the number of tokens or the percentage of total token supply.
              </h3>
              <FormModal
                handleClose={[Function]}
                isOpen={false}
                restrictionType={null}
              />
              <Grid
                gridGap={4}
              >
                <GridRow
                  gridGap={4}
                >
                  <GridCol
                    gridSpan={4}
                  >
                    <CardFeatureState
                      IconAsset={[Function]}
                      maxWidth="none"
                      status="inactive"
                      style={
                        Object {
                          "height": "400px",
                        }
                      }
                    >
                      <Heading__StyledHeading
                        as="h2"
                        color="primary"
                        lineHeight="normal"
                        mb="m"
                        mt={2}
                        variant="h3"
                      >
                        24h Rolling Period Restriction
                      </Heading__StyledHeading>
                      <p
                        className="card-text"
                      >
                        Configure a maximum number of tokens any investor may be able to sell within a rolling 24h period.
                      </p>
                      <div
                        className="restriction-inactive"
                      />
                      <ButtonLarge
                        className="card-button"
                        onClick={[Function]}
                        variant="primary"
                      >
                        Configure 24h Restriction
                      </ButtonLarge>
                    </CardFeatureState>
                  </GridCol>
                  <GridCol
                    gridSpan={4}
                  >
                    <CardFeatureState
                      IconAsset={[Function]}
                      maxWidth="none"
                      status="idle"
                      style={
                        Object {
                          "height": "400px",
                        }
                      }
                    >
                      <Heading__StyledHeading
                        as="h2"
                        color="primary"
                        lineHeight="normal"
                        mb="m"
                        mt={2}
                        variant="h3"
                      >
                        Custom Rolling Period Restriction
                      </Heading__StyledHeading>
                      <p
                        className="card-text"
                      >
                        Configure a maximum number of tokens any Investor may be able to sell within a custom rolling period.
                      </p>
                      <RestrictionDetails
                        restriction={
                          Object {
                            "allowedTokens": 500,
                            "endTime": "2020-12-25T05:00:00.000Z",
                            "restrictionType": 0,
                            "startTime": "2019-12-25T05:00:00.000Z",
                          }
                        }
                      />
                      <div
                        className="restriction-active"
                      />
                      <ButtonLarge
                        className="card-button"
                        onClick={[Function]}
                        variant="primary"
                      >
                        Configure Custom Restriction
                      </ButtonLarge>
                      <ButtonLarge
                        className="card-button"
                        onClick={[Function]}
                        variant="ghostPrimary"
                      >
                        Reset the Restriction
                      </ButtonLarge>
                    </CardFeatureState>
                  </GridCol>
                </GridRow>
              </Grid>
            </Tab>
            <Tab
              href="#"
              label="Individual Restrictions"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={false}
              tabIndex={0}
            >
              <IndividualRestrictions />
            </Tab>
          </Tabs>
        </GridCol>
      </GridRow>
    </Grid>
  </Page>
</ThemeProvider>
`;

exports[`Restrictions Page should render global default restrictions when VRTM is enabled 1`] = `
<ThemeProvider>
  <Page
    py="xl"
    title="Compliance – Polymath"
  >
    <h1
      className="pui-h1"
    >
      Trade Volume Restrictions
    </h1>
    <Grid
      gridGap={4}
    >
      <GridRow
        gridGap={4}
      >
        <GridCol
          gridSpan={6}
        >
          <h3
            className="pui-h3"
          >
            Allows you to define the maximum number of tokens your investors can sell during a customized rolling period. Limits can be set globally for all investors or individually by investor. For example, you may limit all your investors to selling 100 tokens in a rolling 24-hour period and 1,000 tokens in a rolling 90-day period.
          </h3>
          <Remark
            title="Note"
          >
            Investors to whom individual limits are applied are only subject to their individual limits and are exempt from the global limits.
          </Remark>
        </GridCol>
        <GridCol
          gridSpan={6}
        >
          <CardFeatureState
            IconAsset={[Function]}
            maxWidth={355}
            status="idle"
            style={Object {}}
          >
            <Heading__StyledHeading
              as="h2"
              color="primary"
              lineHeight="normal"
              mb="m"
              mt={2}
              variant="h3"
            >
              Enable Enforcement of Trade Volume Restrictions
            </Heading__StyledHeading>
            <div
              className="center"
            >
              <Toggle
                defaultToggled={false}
                id="volumeRestrictionToggle"
                labelA="Off"
                labelB="On"
                onToggle={[Function]}
                toggled={true}
              />
            </div>
          </CardFeatureState>
        </GridCol>
      </GridRow>
      <GridRow
        gridGap={4}
      >
        <GridCol
          gridSpan={12}
        >
          <Tabs
            ariaLabel="listbox"
            iconDescription="show menu options"
            role="navigation"
            selected={0}
            triggerHref="#"
          >
            <Tab
              href="#"
              label="Global Restrictions"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={false}
              tabIndex={0}
            >
              <h1
                className="pui-h1"
              >
                Global Restrictions
              </h1>
              <h3
                className="pui-h3"
              >
                Specify a single rolling period restriction for all investors. The Global Restriction will enforce a maximum number of tokens each investor can sell during the rolling period you define. Easily set to 24 hours or customize your rolling period. Your trade volume cap is specified as the number of tokens or the percentage of total token supply.
              </h3>
              <FormModal
                handleClose={[Function]}
                isOpen={false}
                restrictionType={null}
              />
              <Grid
                gridGap={4}
              >
                <GridRow
                  gridGap={4}
                >
                  <GridCol
                    gridSpan={4}
                  >
                    <CardFeatureState
                      IconAsset={[Function]}
                      maxWidth="none"
                      status="idle"
                      style={
                        Object {
                          "height": "400px",
                        }
                      }
                    >
                      <Heading__StyledHeading
                        as="h2"
                        color="primary"
                        lineHeight="normal"
                        mb="m"
                        mt={2}
                        variant="h3"
                      >
                        24h Rolling Period Restriction
                      </Heading__StyledHeading>
                      <p
                        className="card-text"
                      >
                        Configure a maximum number of tokens any investor may be able to sell within a rolling 24h period.
                      </p>
                      <RestrictionDetails
                        restriction={
                          Object {
                            "allowedTokens": 500,
                            "endTime": "2020-12-25T05:00:00.000Z",
                            "restrictionType": 0,
                            "startTime": "2019-12-25T05:00:00.000Z",
                          }
                        }
                      />
                      <div
                        className="restriction-active"
                      />
                      <ButtonLarge
                        className="card-button"
                        onClick={[Function]}
                        variant="primary"
                      >
                        Configure 24h Restriction
                      </ButtonLarge>
                      <ButtonLarge
                        className="card-button"
                        onClick={[Function]}
                        variant="ghostPrimary"
                      >
                        Reset the Restriction
                      </ButtonLarge>
                    </CardFeatureState>
                  </GridCol>
                  <GridCol
                    gridSpan={4}
                  >
                    <CardFeatureState
                      IconAsset={[Function]}
                      maxWidth="none"
                      status="inactive"
                      style={
                        Object {
                          "height": "400px",
                        }
                      }
                    >
                      <Heading__StyledHeading
                        as="h2"
                        color="primary"
                        lineHeight="normal"
                        mb="m"
                        mt={2}
                        variant="h3"
                      >
                        Custom Rolling Period Restriction
                      </Heading__StyledHeading>
                      <p
                        className="card-text"
                      >
                        Configure a maximum number of tokens any Investor may be able to sell within a custom rolling period.
                      </p>
                      <div
                        className="restriction-inactive"
                      />
                      <ButtonLarge
                        className="card-button"
                        onClick={[Function]}
                        variant="primary"
                      >
                        Configure Custom Restriction
                      </ButtonLarge>
                    </CardFeatureState>
                  </GridCol>
                </GridRow>
              </Grid>
            </Tab>
            <Tab
              href="#"
              label="Individual Restrictions"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={false}
              tabIndex={0}
            >
              <IndividualRestrictions />
            </Tab>
          </Tabs>
        </GridCol>
      </GridRow>
    </Grid>
  </Page>
</ThemeProvider>
`;

exports[`Restrictions Page should render global restrictions tab when VRTM is enabled 1`] = `
<ThemeProvider>
  <Page
    py="xl"
    title="Compliance – Polymath"
  >
    <h1
      className="pui-h1"
    >
      Trade Volume Restrictions
    </h1>
    <Grid
      gridGap={4}
    >
      <GridRow
        gridGap={4}
      >
        <GridCol
          gridSpan={6}
        >
          <h3
            className="pui-h3"
          >
            Allows you to define the maximum number of tokens your investors can sell during a customized rolling period. Limits can be set globally for all investors or individually by investor. For example, you may limit all your investors to selling 100 tokens in a rolling 24-hour period and 1,000 tokens in a rolling 90-day period.
          </h3>
          <Remark
            title="Note"
          >
            Investors to whom individual limits are applied are only subject to their individual limits and are exempt from the global limits.
          </Remark>
        </GridCol>
        <GridCol
          gridSpan={6}
        >
          <CardFeatureState
            IconAsset={[Function]}
            maxWidth={355}
            status="idle"
            style={Object {}}
          >
            <Heading__StyledHeading
              as="h2"
              color="primary"
              lineHeight="normal"
              mb="m"
              mt={2}
              variant="h3"
            >
              Enable Enforcement of Trade Volume Restrictions
            </Heading__StyledHeading>
            <div
              className="center"
            >
              <Toggle
                defaultToggled={false}
                id="volumeRestrictionToggle"
                labelA="Off"
                labelB="On"
                onToggle={[Function]}
                toggled={true}
              />
            </div>
          </CardFeatureState>
        </GridCol>
      </GridRow>
      <GridRow
        gridGap={4}
      >
        <GridCol
          gridSpan={12}
        >
          <Tabs
            ariaLabel="listbox"
            iconDescription="show menu options"
            role="navigation"
            selected={0}
            triggerHref="#"
          >
            <Tab
              href="#"
              label="Global Restrictions"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={false}
              tabIndex={0}
            >
              <h1
                className="pui-h1"
              >
                Global Restrictions
              </h1>
              <h3
                className="pui-h3"
              >
                Specify a single rolling period restriction for all investors. The Global Restriction will enforce a maximum number of tokens each investor can sell during the rolling period you define. Easily set to 24 hours or customize your rolling period. Your trade volume cap is specified as the number of tokens or the percentage of total token supply.
              </h3>
              <FormModal
                handleClose={[Function]}
                isOpen={false}
                restrictionType={null}
              />
              <Grid
                gridGap={4}
              >
                <GridRow
                  gridGap={4}
                >
                  <GridCol
                    gridSpan={4}
                  >
                    <CardFeatureState
                      IconAsset={[Function]}
                      maxWidth="none"
                      status="inactive"
                      style={
                        Object {
                          "height": "400px",
                        }
                      }
                    >
                      <Heading__StyledHeading
                        as="h2"
                        color="primary"
                        lineHeight="normal"
                        mb="m"
                        mt={2}
                        variant="h3"
                      >
                        24h Rolling Period Restriction
                      </Heading__StyledHeading>
                      <p
                        className="card-text"
                      >
                        Configure a maximum number of tokens any investor may be able to sell within a rolling 24h period.
                      </p>
                      <div
                        className="restriction-inactive"
                      />
                      <ButtonLarge
                        className="card-button"
                        onClick={[Function]}
                        variant="primary"
                      >
                        Configure 24h Restriction
                      </ButtonLarge>
                    </CardFeatureState>
                  </GridCol>
                  <GridCol
                    gridSpan={4}
                  >
                    <CardFeatureState
                      IconAsset={[Function]}
                      maxWidth="none"
                      status="inactive"
                      style={
                        Object {
                          "height": "400px",
                        }
                      }
                    >
                      <Heading__StyledHeading
                        as="h2"
                        color="primary"
                        lineHeight="normal"
                        mb="m"
                        mt={2}
                        variant="h3"
                      >
                        Custom Rolling Period Restriction
                      </Heading__StyledHeading>
                      <p
                        className="card-text"
                      >
                        Configure a maximum number of tokens any Investor may be able to sell within a custom rolling period.
                      </p>
                      <div
                        className="restriction-inactive"
                      />
                      <ButtonLarge
                        className="card-button"
                        onClick={[Function]}
                        variant="primary"
                      >
                        Configure Custom Restriction
                      </ButtonLarge>
                    </CardFeatureState>
                  </GridCol>
                </GridRow>
              </Grid>
            </Tab>
            <Tab
              href="#"
              label="Individual Restrictions"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="presentation"
              selected={false}
              tabIndex={0}
            >
              <IndividualRestrictions />
            </Tab>
          </Tabs>
        </GridCol>
      </GridRow>
    </Grid>
  </Page>
</ThemeProvider>
`;

exports[`Restrictions Page should render not found when there is no token 1`] = `<NotFoundPage />`;

exports[`Restrictions Page should render the page with VRTM disabled 1`] = `
<ThemeProvider>
  <Page
    py="xl"
    title="Compliance – Polymath"
  >
    <h1
      className="pui-h1"
    >
      Trade Volume Restrictions
    </h1>
    <Grid
      gridGap={4}
    >
      <GridRow
        gridGap={4}
      >
        <GridCol
          gridSpan={6}
        >
          <h3
            className="pui-h3"
          >
            Allows you to define the maximum number of tokens your investors can sell during a customized rolling period. Limits can be set globally for all investors or individually by investor. For example, you may limit all your investors to selling 100 tokens in a rolling 24-hour period and 1,000 tokens in a rolling 90-day period.
          </h3>
          <Remark
            title="Note"
          >
            Investors to whom individual limits are applied are only subject to their individual limits and are exempt from the global limits.
          </Remark>
        </GridCol>
        <GridCol
          gridSpan={6}
        >
          <CardFeatureState
            IconAsset={[Function]}
            maxWidth={355}
            status="inactive"
            style={Object {}}
          >
            <Heading__StyledHeading
              as="h2"
              color="primary"
              lineHeight="normal"
              mb="m"
              mt={2}
              variant="h3"
            >
              Enable Enforcement of Trade Volume Restrictions
            </Heading__StyledHeading>
            <div
              className="center"
            >
              <Toggle
                defaultToggled={false}
                id="volumeRestrictionToggle"
                labelA="Off"
                labelB="On"
                onToggle={[Function]}
              />
            </div>
          </CardFeatureState>
        </GridCol>
      </GridRow>
      <GridRow
        gridGap={4}
      >
        <GridCol
          gridSpan={12}
        />
      </GridRow>
    </Grid>
  </Page>
</ThemeProvider>
`;
